### DISCUSSION TITLE
How should cookies be handled, frontend or backend?

### DISCUSSION THREAD

**Discussion by BeeMoe5** [fastapi/fastapi#4256] [Questions]
Context is I use to have just fastapi but now I'm integrating the app with react for better frontend but at this point I had a lot setup so I don't wanna reset my progress but in the end if it's better to use frontend I will. Bonus question: is using the react proxy the only way to connect the frontend to the backend?

**Comment by simondale00** [comment]
Hey @minecrosters. Can you clarify what you mean by "handling" cookies? Like, creating them, reading them, etc?

Best,
Simon

**Reply by BeeMoe5** [reply]
Yes, writing and reading

**Reply by simondale00** [reply]
Hey @minecrosters.

Ultimately I think it all depends on your use-case.

Many apps use cookies to represent user sessions. In this case you want to ensure that sessions are secure, so only the server should be writing and reading the cookie value.

Some cookies may store arbitrary data; feature flags, tracking data, etc. Security may not be a concern here, so you could have either the frontend or the backend handle the cookies. Though keep in mind that a browser with automatically send _all_ cookies for the domain to your backend. 

If you have a concrete use-case in mind I can provide more specific advice.

Best,
Simon

**Reply by BeeMoe5** [reply]
@simondale00 the first case is my case actually, using SessionMiddleware on fastapi. Since the react frontend is new to my app it's difficult for me to integrate it as I am also new to react. Since the frontend is running on a different ip/port than the backend I don't know what I could do in this situation. I'm using a "login with discord" module made for starlette and storing the refresh token in with the SessionMiddleware (assuming that cookies are okay for that, if not let me know the alternative way)

Another thing is, with react in package.json I have the proxy pointed to the backend and when I make a request to the backend, the fastapi logging has None where an ip/port should be.

**Comment by ahmed0saber** [comment]
that depends upon what that Cookies store. if something like login data, user score in a game, profile info like followers and these things, use Backend for handling Cookies...
But treating with online/offline status, draft/unsaved changes, profile picture for offline displaying with browser cache. use Frontend...
that's just my opinion !
